### PR TITLE
Added support of rendering math, mi, mn, math, mrow tags.

### PR DESCRIPTION
**Pull request by vishwesh3**
on [current master](https://github.com/zulip/zulip-mobile/tree/c02dba3c8081b218b8826b544072fd08c2f7e543)
screen shot 2017-07-01 at 7 38 06 pm

This PR
on Web
screen shot 2017-07-01 at 7 35 52 pm

on mobile
screen shot 2017-07-01 at 7 32 17 pm

message link https://chat.zulip.org/#narrow/stream/exams.20study.20group/subject/math/near/239034

### CODE REVIEW

**Comment by saketkumar95** [comment]
@vishwesh3 Awesome! I was also on this issue .. 👍

**Comment by borisyankov** [comment]
Can you give a link to Yago's message that has this rendering.
I want to investigate what is the html related.

**Comment by vishwesh3** [comment]
message link https://chat.zulip.org/#narrow/stream/exams.20study.20group/subject/math/near/239034

html
```
<p>It is explained in the formatting help overlay:</p>
<blockquote>
   <p>
      Some inline math 
      <span class="katex">
         <span class="katex-mathml">
            <math>
               <semantics>
                  <mrow>
                     <msup>
                        <mi>e</mi>
                        <mrow>
                           <mi>i</mi>
                           <mi>π</mi>
                        </mrow>
                     </msup>
                     <mo>+</mo>
                     <mn>1</mn>
                     <mo>=</mo>
                     <mn>0</mn>
                  </mrow>
                  <annotation encoding="application/x-tex">e^{i \pi } + 1 = 0</annotation>
               </semantics>
            </math>
         </span>
         <span aria-hidden="true" class="katex-html"><span class="strut" style="height:0.824664em;"></span><span class="strut bottom" style="height:0.907994em;vertical-align:-0.08333em;"></span><span class="base textstyle uncramped"><span class="mord"><span class="mord mathit">e</span><span class="msupsub"><span class="vlist"><span style="top:-0.363em;margin-right:0.05em;"><span class="fontsize-ensurer reset-size5 size5"><span style="font-size:0em;">​</span></span><span class="reset-textstyle scriptstyle uncramped mtight"><span class="mord scriptstyle uncramped mtight"><span class="mord mathit mtight">i</span><span class="mord mathit mtight" style="margin-right:0.03588em;">π</span></span></span></span><span class="baseline-fix"><span class="fontsize-ensurer reset-size5 size5"><span style="font-size:0em;">​</span></span>​</span></span></span></span><span class="mbin">+</span><span class="mord mathrm">1</span><span class="mrel">=</span><span class="mord mathrm">0</span></span></span>
      </span>
   </p>
</blockquote>
```

File: src/html/HtmlNodeTag.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { Auth, Message, PushRouteAction } from '../types';
 import styles from './HtmlStyles';
 import cascadingStyles from './cascadingStylesView';
 import cascadingStylesText from './cascadingStylesText';
+import msup from './indexBasedStyles/msup';
```
Comment (borisyankov) [inline src/html/HtmlNodeTag.js]: You can just import directly like:
```js
import indexBasedStyles from './indexBasedStyles';
```
and then have `indexBasedStyles.js` file with
```js
export default {
  msup: StyleSheet.create({
  })
}
```

File: src/html/renderHtmlChildren.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ type Props = {
   childrenNodes: ?any[],
   cascadingStyle?: StyleObj,
   cascadingTextStyle?: StyleObj,
+  indexBasedStyles?: any[],
```
Comment (borisyankov) [inline src/html/renderHtmlChildren.js]: Maybe just call `indexedStyles`?

**Comment by borisyankov** [comment]
Please, make the changes, rebase the eslintrc issue and it is ready to merge.

File: src/html/renderHtmlChildren.js (hunk 2 above)
Comment (vishwesh3) [inline src/html/renderHtmlChildren.js]: Done 👍

File: src/html/HtmlNodeTag.js (hunk 1 above)
Comment (vishwesh3) [inline src/html/HtmlNodeTag.js]: Done 👍

**Comment by vishwesh3** [comment]
@borisyankov updated 
ready to go in 🏃

**Comment by borisyankov** [comment]
👍

**Pull request merged**